### PR TITLE
Developer UX: Add Hosting Configuration item with upsell info

### PR DESCRIPTION
**Pull request by sejas**
- Closes https://github.com/Automattic/dotcom-forge/issues/1746

## Proposed Changes

* Rename `Developer settings` to `Hosting configuration`
* Take to the top level the thee subitems: `Performance settings`, `Privacy settings` and `Domains and DNS`
* Add upsell information in hosting configuration submenu items and manage plugins.
* Each upsell checks for different site features (Plugins | SFTP)
* Hide Plugins for P2 sites

## Testing Instructions

* Go to `/sites`
* Open the actions menu of a free site
* Observe the info description says `Requires a Business Plan`
* Open the actions menu of a business site
* Observe there is no additional info.

## Screenshots

| **Free** | **Business** |
|---|---|
| Screenshot 2023-03-01 at 17 47 52 | Screenshot 2023-03-01 at 17 52 58 |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by sejas** [comment]
@javierarce What do you think?
Do you like that we show different texts if it's included in the plan vs not?

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 1]
```diff
@@ -375,15 +347,47 @@ export const SitesEllipsisMenu = ( {
...
+						</MenuItemLink>
+					) }
+					{ hasCustomDomain && (
+						<MenuItemLink
+							href={ `/domains/manage/${ site.slug }/dns/${ site.slug }` }
+							onClick={ () =>
+								recordTracks( 'calypso_sites_dashboard_site_action_submenu_dns_records_click' )
+							}
+						>
+							{ __( 'DNS records' ) }
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]:
```suggestion
							{ __( 'Domains and DNS' ) }
```

File: client/sites-dashboard/hooks/use-upsell-info.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { useI18n } from '@wordpress/react-i18n';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+
+export function useUpsellInfo( site: SiteExcerptData ): string {
+	const { __ } = useI18n();
+	return site.is_wpcom_atomic ? __( 'Included in your plan' ) : __( 'Requires a Business Plan' );
```
Comment (danielbachhuber) [inline client/sites-dashboard/hooks/use-upsell-info.ts]: I think 'Included in your plan' is a little bit much:

image

AFAIK, we don't have that pattern elsewhere in the UI.

Also, a Simple site may have a Business plan so `site.is_wpcom_atomic` isn't necessarily the most appropriate check here.

What about something like this? Maybe with an icon?

image

Comment (sejas) [reply client/sites-dashboard/hooks/use-upsell-info.ts]: > Also, a Simple site may have a Business plan so site.is_wpcom_atomic isn't necessarily the most appropriate check here.

Yes!, I need to improve it. I'll check if the site has the SFT feature available.

> What about something like this? Maybe with an icon?

Yes!, I envision it to be something similar to the jetpack icon.

Comment (javierarce) [reply client/sites-dashboard/hooks/use-upsell-info.ts]: What do you think about doing something like this instead? The button would direct the customer to the Hosting Configuration page that shows the features included.

image

**Comment by javierarce** [comment]
> Do you like that we show different texts if it's included in the plan vs not?

I think we should only mention that certain features are available after an upgrade, not those that are already included in the customer's plan.

**Comment by sejas** [comment]
> > Do you like that we show different texts if it's included in the plan vs not?
> 
> I think we should only mention that certain features are available after an upgrade, not those that are already included in the customer's plan.

Thanks! I've updated the PR hiding the upsell `info` when the site has the feature. I want to work a bit more on this PR to try a new approach, displaying only the badge start.

File: client/sites-dashboard/hooks/use-upsell-info.ts (hunk 2 above)
Comment (sejas) [reply client/sites-dashboard/hooks/use-upsell-info.ts]: I want to suggest another alternative. We display the star icon for free plans and the  `Requires Business Plan` text when the mouse hovers over the icon to explain what it means.
We use the `info` menu item text to increase awareness of the dev features, using geek terms: `SSH, DDBB, GitHub …`

@javierarce , what do you think?

- I've created this PR to play with this solution https://github.com/Automattic/wp-calypso/pull/73934
- This is a rough implementation. We can improve the design even more.

## Screencast:

https://user-images.githubusercontent.com/779993/222123967-38aec15c-957b-4f7a-aabc-c6639bf4730f.mp4



## Screenshots

| **Free** | **Business** |
|---|---|
| Screenshot 2023-03-01 at 11 15 13 | Screenshot 2023-03-01 at 11 15 13 |

Comment (javierarce) [reply client/sites-dashboard/hooks/use-upsell-info.ts]: I'm afraid I don't like this solution. Even if we tweaked the design, the use of the icon is very redundant and requires you hover/select the item to read the upsell message.

Comment (sejas) [reply client/sites-dashboard/hooks/use-upsell-info.ts]: > image

My only concern about this approach is that we are not applying the same pattern for `Plugins`, which requires a Business plan too.

Comment (sejas) [reply client/sites-dashboard/hooks/use-upsell-info.ts]: A less repetitive option, without showing the submenu for free sites.

@javierarce, let's talk in sync to find the best solution 💪 

| **Free** | **Business** |
|---|---|
| Screenshot 2023-03-01 at 14 22 26 |  |

**Comment by sejas** [comment]
@javierarce I've updated the PR with the last design we've discussed 👌 

Thanks for all your help!

Screenshot 2023-03-01 at 17 47 52

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 3]
```diff
@@ -246,102 +257,84 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
...
+				href: `/hosting-config/${ siteSlug }#sftp-credentials`,
+				eventName: 'calypso_sites_dashboard_site_action_submenu_sftp_credentials_click',
+			},
+			{
+				label: __( 'Database access' ),
+				href: `/hosting-config/${ siteSlug }#database-access`,
+				eventName: 'calypso_sites_dashboard_site_action_submenu_database_access_click',
+			},
+			{
+				label: __( 'Github connection' ),
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]:
```suggestion
				label: __( 'Deploy from GitHub' ),
```

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 4]
```diff
@@ -246,102 +257,84 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
...
+				href: `/hosting-config/${ siteSlug }#connect-github`,
+				eventName: 'calypso_sites_dashboard_site_action_submenu_connect_github_click',
+			},
+			{
+				label: __( 'Web server settings' ),
+				href: `/hosting-config/${ siteSlug }#web-server-settings`,
+				eventName: 'calypso_sites_dashboard_site_action_submenu_web_server_settings_click',
+			},
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx:285]: Should we add 'Web server logs' here?

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 5]
```diff
@@ -246,102 +257,84 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
...
+							href={ item.href }
+							onClick={ () => recordTracks( item.eventName ) }
+						>
+							{ item.label }
+						</MenuItemLink>
+					) )
+				) : (
+					<UpsellMenuGroup>
+						{ __(
+							'Upgrade to Business Plan to enable SFTP & SSH, database access, GitHub connection and more …'
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]:
```suggestion
							'Upgrade to the Business Plan to enable SFTP & SSH, database access, GitHub deploys, and more…'
```

**Review by danielbachhuber — APPROVED** [review]
I love where this ended up — great work!

Left a few minor comments. Generally, it seems like there's some complexity developing around which menu items should appear in which context. Maybe we should have some test coverage around that?

Here are the scenarios I evaluated:

## Free site

image

## Personal site

image

## Business site on Simple

image

## Redirect site

image

## Business site

image

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx (hunk 3 above)
Comment (sejas) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx]: Done https://github.com/Automattic/wp-calypso/pull/73824/commits/bb4f2943d6ad828318dc0b8ff115e9e06ecce235

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx (hunk 5 above)
Comment (sejas) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx]: Done https://github.com/Automattic/wp-calypso/pull/73824/commits/705763a83e0f760583b5d555870da676f322f23a

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx (hunk 4 above)
Comment (sejas) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx:285]: Done https://github.com/Automattic/wp-calypso/pull/73824/commits/9c6c56d741cf6cd3ccfa20f0bb5515eabc8a61b3

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7838946

Thank you @sejas for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7838946).

**Pull request merged**